### DISCUSSION TITLE
Playtime: Allow to change tempo when triggering a scene

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#1042] [Ideas]

**Comment by EachDawn** [comment]
Along these lines, it could also be cool if Playtime supported tempo ranges (e.g. 80 ->100 BPM) within a Scene.

**Reply by DerWhistler** [reply]
Having the possibility to change tempo with scenes is crucial for setting up live performances.
You have different songs with different tempos that have to be pre-defined and should work by starting a scene.
I know that Playtime2 is tied to Reaper tempo. The only way I can think of to solve this would be tempo markers that one could jump to.
Would it be possible to setup a dummy clip that makes it possible to jump to different positions in an (otherwise empty) arrangement.
Setting up different tempos on different song positions to jump to.
Just for the purpose of changing tempo? This would have other benefits as well. 
It would be possible to use Reaper inbuilt automation to activate/deactivate VSTs on given tracks that get triggered by Playtime2.
This may adress the wish of some users to have midi clock sent to VSTs. I know that it´s not about midi-clock, but this way everything could stay in sync.
This would make static tempo changes possible (funny wording :-)) I mean no seamless accelleration, though.
Also the VSTs would be perfectly in sync.

**Reply by helgoboss** [reply]
Playtime's engine is not tied to REAPER tempo. If I would want, I could make it have a completely different tempo. Following the REAPER tempo turned out to be practical, however (who wants 2 separate tempos anyway). So currently, there's a bidirectional sync between Playtime's tempo and REAPER's project tempo (not envelope tempo currently). Sudden REAPER project tempo changes are picked up by Playtime.

Switching tempos between scenes is totally possible within the engine, because it could be done just like a sudden tempo change. It would work best when doing playback just within Playtime. The issues start when playing together with the REAPER arrangement, because REAPER itself doesn't support sudden tempo changes.

**Reply by DerWhistler** [reply]
Hello Benjamin, 
thank you for your reply. Of course no one want´s 2 separate tempos. ;-)
Ok, I understood: The tempo of PT2 is independent from Reaper but they sync up to each other (can be switched of).
I still don´t understand how I can create tempo changes (for different songs).
I don´t want to play PT2 alongside to Reapers playback engine. Reaper would just be used as a VST host.
I was thinking about markers because it would make switching on/off VSTs possible to save CPU via automation.

Can this be done in the current version of PT2? Changing tempo from scene to scene? 
Maybe I am missing something.

**Comment by helgoboss** [comment]
No, it can't be done yet. I was just talking about the theoretical
capabilities of the engine.

This discussion is about adding it. I think there's another feature request
in this list where someone requested to trigger ReaLearn targets when
switching scenes. That would be an alternative to your marker suggestion
(and one that wouldn't require REAPER playback).


TheWhistler ***@***.***> schrieb am Mi., 28. Aug. 2024, 09:57:

**Comment by DerWhistler** [comment]
Hello Bejamin, please tell how to change tempo from scene to scene. I can´t find how to do this.
Thank you in advance :-)

**Reply by DerWhistler** [reply]
Sorry, misread your comment. I was searching up and down the information that is yet available.
My bad, sorry "can´t be done yet" means can´t be done yet. 
Wishful thinking. I would really need this to get Playtime working for a complete performance, though.
Can I open a feature request (most probably I can)?

**Reply by helgoboss** [reply]
This is the feature request 😆

**Reply by DerWhistler** [reply]
Sorry, laugh at me, at least funny for you....

**Reply by DerWhistler** [reply]
How likely is this to happen?

**Reply by helgoboss** [reply]
Sorry. Was supposed to be just a winking face smiley ;)

This is going to happen. But my top priority right now is bugfixing and polishing, so that I can make the final release soon.

**Reply by DerWhistler** [reply]
No problem, I was laughing at myself as well. To funny I did not grasp that. ;-)
This is good news as I already stated that this is really important for me.

**Reply by DerWhistler** [reply]
Hello Benjamin, is there something new on this topic. I know you are working constantly on Playtime/ReaLearn which is very much appreciated.
Changing tempo on a per scene basis is crucial for me if I want to use Playtime for live use.
I would want to have a constant stream of audio (few tracks only) that would sync to what ever I put aside (Midi clocked synths e.g.).
The pre-recorded audio files would be a backbone to my performance.
Thank you very much and have a good time.
Cheers Mario

**Reply by helgoboss** [reply]
I'm still waiting for some feedback on the current pre-release versions. I'm somewhat hesitant to make them official without enough feedback.

After that, I could give this one a try. In this regard, it would be good to know what is the minimum implementation that would be useful for you:

a) Change tempo instantly to provided value when pressing scene launch => Low effort
b) Change tempo on next bar (or something else, according to quantization) when pressing scene launch => Medium effort
c) Change tempo gradually until next bar => Higher effort

So if you need this feature, I suggest the following procedure:

1. Please give the current pre-release version a try and report back if you find any major issues
2. Choose a, b or c. a could be done quite fast, I could even integrate it into the upcoming stable release.

**Reply by DerWhistler** [reply]
I will give the current beta a try (weekend due to daytime job ;-))
For my needs a simple "change tempo on next scene" would be enough, so solution a) so far.
I am into electronic dance music (kinda experimental sometimes) and the tracks I perform normally have a constant tempo.
This is what my Octatrack does at the moment. Playing stems that get treated with FX (midi and audio).

I would not go as far to say I could not make use of a gradual change in tempo from one scene to another.
If this could be something to happen in the future, I might not be the only one using it (but don´t really know of course). 

I can not think of a situation where I would want to change tempo at the next bar.
The next bar could also be the next scene. Well, with follow actions perhaps. :-)

So thank you for your constant effort on this.
Will report if I find something in the beta.

BTW: I never installed betas because I use Reapack and just sync the repo. 
Is there some special way to install Playtime betas?

**Reply by helgoboss** [reply]
Hi. It's described here: https://docs.helgoboss.org/helgobox/installation.html#pre-releases

**Comment by helgoboss** [comment]
@DerWhistler Will be available in 2.18.0 (#1467).

**Comment by DerWhistler** [comment]
Hello Benjamin,
this is really good news, thank you very much for implementing.
So the next step to make playtime the backbone of my live set will be sync with hardware boxes.
Cheers
